### ISSUE TITLE
Make Gantt graphs printable to pdf

### ISSUE THREAD

**Issue by dult** [kanboard/kanboard#1329]
The ability to print a project Gantt graph to a pdf/excel without it breaking would make it easy to make reports about the current state of project within an organization and with external clients.

As they are currently implemented, Gantt graphs can only be used for consultation.

Now, would it be better to print to PDF or to convert the Gantt to an XLS file?

**Comment by phillc73** [comment]
+1 for this request.

**Comment by SkyBE** [comment]
+1 for this request.

**Comment by milankni** [comment]
+1.
Not only Gantt - I am not able to print any other page with a reasonable layout. Workaround is to display the page in full screen and make a "screenshot". The quality of fonts suffer heavily though...

**Comment by CoMPaTech** [comment]
TL;DR If html2canvas is a feasible solution for you we'll need indication how to find width of ```#gantt-chart``` div so to-be-written JS can temp. override body and slide-container. Rest is coding of a button that does that and fires at html2canvas. @fguillot 

For those subscribed;
I've made some milage on trying this out, but got stuck on one particular thing.
It's sort of making the screenshots, but in a more sophisticated way :)

Main problem would be 'how-to-adjust-the-body-width' to the correct length. I've looked around but didn't find a good answer for it in the code/js. The 4000 was a lucky guess to show my current timeshift in a project. The container overflows are do-able with a print-button, as is the inclusion of the html2canvas code and the actual print javascript.

Problem at hand, you need to make to gantt 'fit' on your screen. You can reproduce all the steps below in your browser using a CSS-inspector and the console.

Adjust CSS for 
  * ```body { width: 4000px; }``` (<- manual action)
  * ```div.ganttview-slide-container { overflow-y: auto; overflow-x: hidden; }```

Include (or paste in browser console)
  * https://html2canvas.hertzen.com - for console, copy/paste the _contents_ from https://raw.githubusercontent.com/niklasvh/html2canvas/master/dist/html2canvas.min.js
  * In your console enter the below, wait for it and print!

```
html2canvas($("#gantt-chart"), {
                onrendered: function (canvas) {
                var myImage = canvas.toDataURL("image/png");
                 var tWindow = window.open(""); 
      $(tWindow.document.body).html("<img id='Image' src=" + myImage + " style='width:100%;'></img>").ready(function () {
         tWindow.focus();
           tWindow.print();
               });
                }
            })
```

Sample (partial) safari print preview (spanning feb-aug of this year):
![kanboard gantt print](https://cloud.githubusercontent.com/assets/5751088/14412914/60699fe8-ff6c-11e5-96a2-a9303552d8cf.png)

**Comment by fguillot** [comment]
@CoMPaTech Nice! I didn't even know that html2canvas existed.

**Comment by CoMPaTech** [comment]
Didn't know either, just found out by trying different things. The solution as pasted above works, if you concatenate the html2canvas and the additional javascript it already works and prints the gantt 'as you see it'.
But I think most users would like their whole gantt and not only the visible part. Thats why I put in the 4000px (but that just matches one board, not all)

**Comment by kgolding** [comment]
+1 for this and for better print layout on all pages!

**Comment by mpoulain** [comment]
Thanks for ur work @fguillot & @CoMPaTech u save a lot of time to me and contribute to writing my studying report and u give me the possibility does not use M$Project :smile: 
Do you think you add "native" feature in future version?

**Comment by asmwarrior** [comment]
+1 for this feature request.

**Comment by fguillot** [comment]
Please, move your ticket to the Gantt plugin bug tracker: https://github.com/kanboard/plugin-gantt/issues

- This will facilitate the work of the maintainers (bug triage)
- Make sure your ticket still relevant with the latest version of the software
- If you are asking another feature request, do not hesitate to contribute, Kanboard is an open source project. Remember the famous quote from Linus Torvalds: _“Talk is cheap. Show me the code.”_

Thank you for your understanding

**Comment by felie** [comment]
Dont reinvent the wheel. https://getfireshot.com/ 
Kanboard is a marvelous tool. 
Best regards
